### PR TITLE
[Sync] Force set sync decryption passphrase

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
When sync decryption passphrase is not set during sync setup, then sync goes into error state with bunch of `GenerateCryptoErrorForTypes` errors. This PR does set decryption passphrase for such cases.

Resolves https://github.com/brave/brave-browser/issues/22898

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket #22898](https://github.com/brave/brave-browser/issues/22898) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
See the issue https://github.com/brave/brave-browser/issues/22898, `Steps to reproduce if you can't run local sync server:` section.

### CODE REVIEW

File: components/sync/driver/brave_sync_service_impl.cc [hunk 1]
```diff
@@ -142,4 +142,33 @@ void BraveSyncServiceImpl::ResumeDeviceObserver() {
...
+  }
+
+  syncer::SyncUserSettings* sync_user_settings = GetUserSettings();
+  if (!sync_user_settings->IsFirstSetupComplete()) {
+    return;
+  }
+
+  bool failed_to_decrypt = false;
+  std::string passphrase = brave_sync_prefs_.GetSeed(&failed_to_decrypt);
+  if (passphrase.empty()) {
```
Comment (darkdh) [inline components/sync/driver/brave_sync_service_impl.cc:163]: we should also check failed_to_decrypt here

File: components/sync/driver/brave_sync_service_impl_unittest.cc [hunk 2]
```diff
@@ -98,6 +118,8 @@ class BraveSyncServiceImplTest : public testing::Test {
     return component_factory()->last_created_engine();
   }
 
+  void run_until_idle() { task_environment_.RunUntilIdle(); }
```
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply components/sync/driver/brave_sync_service_impl_unittest.cc]: thanks, this is fixed at https://github.com/brave/brave-core/compare/854d061546e5daf9b3a90267e1c752def2c0375f..ee92ac04ab82e041dd5170bd2e5b0512db57610c

File: components/sync/driver/brave_sync_service_impl.cc (hunk 1 above)
Comment (AlexeyBarabash) [reply components/sync/driver/brave_sync_service_impl.cc:164]: I have added `DCHECK` at https://github.com/brave/brave-core/compare/854d061546e5daf9b3a90267e1c752def2c0375f..ee92ac04ab82e041dd5170bd2e5b0512db57610c .

I admit, I can't do more. If `passphrase` is empty, and `failed_to_decrypt` is `true` - this means that we are on macOS and we could not get access to the keychain  => we cannot get plain `passphrase` and we cannot pass it into `sync_user_settings->SetDecryptionPassphrase` even if we understand that we need to pass it.

Comment (darkdh) [reply components/sync/driver/brave_sync_service_impl.cc:164]: actually I was referring to `passphrase.empty() || failed_to_decrypt`

Comment (AlexeyBarabash) [reply components/sync/driver/brave_sync_service_impl.cc:164]: but if `failed_to_decrypt` is true, then `passphrase` is always empty

Comment (darkdh) [reply components/sync/driver/brave_sync_service_impl.cc:164]: ok, then DCHECK looks good

**Review by darkdh — APPROVED** [review]

**Comment by kjozwiak** [comment]
Reproduced the original issue using `1.39.115 Chromium: 102.0.5005.78` using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/22898.

**`Example`** | **`Example`** | **`Example`** | **`Example`**
---------------|----------------|-----------------|---------------
Screenshot_20220606-225844 | Screenshot_20220606-230111 | Screenshot_20220606-230514 | Screenshot_20220606-230455

Verification **PASSED** on `Pixel 6` running `Android 13 BETA` using the following build(s):

```
Brave | 1.41.41 Chromium: 102.0.5005.78 (Official Build) canary (32-bit)
--- | ---
Revision | df6dbb5a9fd82af3f567198af2eb5fb4876ef99c-refs/branch-heads/5005_59@{#3}
OS | Android 12; Build/TPBB.220414.018
```

* ensured that once the internet has been disabled, the device is added into the sync chain without any issues
* ensured that syncing is working as expected (example: ensure removing/adding bookmarks is being synced to all devices)
* ensured that the `GenerateCryptoErrorForTypes` error isn't visible via `brave://sync-internals`

**`Example`** | **`Example`** | **`Example`**
---------------|----------------|------------------
sync | Screenshot_20220607-000104 | Screenshot_20220606-234653

**Comment by DaniloRoque501** [comment]
gh pr checkout 13397

**Pull request merged**